ParameterDataApi.yaml

```diff
@@ -88,122 +144,79 @@ components:
           type: string
           format: date-time
-            
-        fields:
-          type: array
-          items:
-            $ref: '#/components/schemas/ValueSpec'
-            
+          description: The timestamp that applies to the measurements in the data field (DataSample)
+          
+        data: 
+          description: A row of one or more data values, associated with a single discrete timestamp. Key/value pairs should be used, including units in the key name. The fields within this object should always follow a consistent order from row to row given a combination of messageType and version
+          type: object
+          additionalProperties: true    
+        
       required:
         - timestamp
-        - fields
+        - data    
 
-    DataType:
-      type: string
-      enum:
-        - NUMBER
-        - INTEGER
-        - STRING
-        
     HeaderDefinition:
       type: object
       properties: 
-        startTime:
-          description: The start time for the time series
+      
+        componentUuid:
           type: string
-          format: date-time
-        
-        endTime: 
-          description: The end time for the time series
+          format: uuid
+          description: The unique identifier of the component within FRAIHMWORK that the parameter data associates with
+      
+        messageType:
+          type: string          
+          description: The type of the message, dictated by the client for delineating individual message formats
+
+        messageVersion:
           type: string
-          format: date-time
+          description: The version string for the parameter data messageType. If the client changes the data fields within a message stream, they should increment the version accordingly to distinguish message format along with the messageType
           
-        columns:
-          $ref: '#/components/schemas/ColumnDefinition'
-        
       required:
-        - startTime
-        - endTime
-        - columns
+        - componentUuid
+        - messageType
+        - version
 
-    TimeSeriesDataDefinition:
+    SingleDataDefinition:
       type: object
-      description: A definition of time series data. This is loosely based upon Json Time Series, Version 1.0. (https://docs.eagle.io/en/latest/reference/index.html)
+      description: A definition of a set of parameter data having a single discrete timestamp. For multiple parameter observations for the given timestamp, use multiple key/value pairs with unique keys
       properties:
-        docType:
-          type: string          
-          description: The type of the document
-
-        version:
-          type: string
-          description: The version string for the time series data definition
 
         header:
           $ref: '#/components/schemas/HeaderDefinition'
 
-        data:
-          $ref: '#/components/schemas/DataDefinition'
-
+        single:
```

Comment:
Was this added to have parity with the series data type, or just to avoid duplication of the DataSample fields? If the former, I wonder if this is introducing an unneeded layer of nesting, since these two endpoints are unlikely to be used by the same client, and thus I don't think they'll care, but curious for your take on it @MS-ResilienX. If it is the latter, this is solvable through schema definition (allOf keyword).

Reply:
allOf brings the data and timestamp fields up to the same level. This is a good recommendation, we don't need hard consistency between the structure of the two data types for the endpoints.